crawler/filtered_positive_identifier.py

```diff
@@ -44,23 +42,58 @@ def is_positive(self, sample: Sample) -> bool:
         if self.result_regex.match(sample[FIELD_RESULT]) is None:
             return False
 
-        if self.root_sample_id_control_regex.match(sample[FIELD_ROOT_SAMPLE_ID]) is not None:
+        if (
+            self.root_sample_id_control_regex
+            and self.root_sample_id_control_regex.match(sample[FIELD_ROOT_SAMPLE_ID]) is not None
+        ):
             return False
 
-        ch1_cq = sample.get(FIELD_CH1_CQ)
-        ch2_cq = sample.get(FIELD_CH2_CQ)
-        ch3_cq = sample.get(FIELD_CH3_CQ)
+        if self.evaluate_ct_values:
+            ch1_cq = sample.get(FIELD_CH1_CQ)
+            ch2_cq = sample.get(FIELD_CH2_CQ)
+            ch3_cq = sample.get(FIELD_CH3_CQ)
+
+            if ch1_cq is None and ch2_cq is None and ch3_cq is None:
+                return True
 
-        if ch1_cq is None and ch2_cq is None and ch3_cq is None:
+            with decimal.localcontext(self.d128_context):
+                # type check before attempting to convert to decimal
+                if ch1_cq is not None and ch1_cq.to_decimal() <= self.ct_value_limit:
+                    return True
+                elif ch2_cq is not None and ch2_cq.to_decimal() <= self.ct_value_limit:
+                    return True
+                elif ch3_cq is not None and ch3_cq.to_decimal() <= self.ct_value_limit:
+                    return True
+                else:
+                    return False
+        else:
             return True
 
-        with decimal.localcontext(self.d128_context):
-            # type check before attempting to convert to decimal
-            if ch1_cq is not None and ch1_cq.to_decimal() <= self.ct_value_limit:
-                return True
-            elif ch2_cq is not None and ch2_cq.to_decimal() <= self.ct_value_limit:
-                return True
-            elif ch3_cq is not None and ch3_cq.to_decimal() <= self.ct_value_limit:
-                return True
-            else:
-                return False
+
+def current_filtered_positive_identifier() -> FilteredPositiveIdentifier:
+    return FilteredPositiveIdentifierV2()
+
+
+class FilteredPositiveIdentifierV0(FilteredPositiveIdentifier):
+    def __init__(self):
+        super(FilteredPositiveIdentifierV0, self).__init__()
+        self.version = FILTERED_POSITIVE_VERSION_0
+        self.result_regex = re.compile(f"^{POSITIVE_RESULT_VALUE}", re.IGNORECASE)
```

Comment:
The `result_regex` is the same for each version. Could be extracted out to the parent class